compiler/resolution/ResolutionCandidate.cpp

```diff
@@ -957,18 +963,18 @@ bool ResolutionCandidate::checkGenericFormals(Expr* ctx) {
         return false;
       }
 
-      if (ConstrainedType* cat = toConstrainedType(actual->getValType()))
-        // a CT actual matches only against itself
-        if (cat != formal->type)
-          // allow stand-in types to match any generic formal at this point
-          if (! (cat->ctUse == CT_GENERIC_STANDIN &&
-                 (formal->type == dtUnknown ||
-                  formal->type == dtAny     ||
-                  formal->type->symbol->hasFlag(FLAG_GENERIC))) ) {
-            failingArgument = actual;
-            reason = RESOLUTION_CANDIDATE_INTERFACE_FORMAL_AS_ACTUAL;
-            return false;
-          }
+      if (ConstrainedType* actCT = toConstrainedType(actual->getValType())) {
+        if (actCT == formal->type)
+          ; // ok: a CG actual matches against the same type
```

Comment:
let's add curly braces around these sections in this if/else if to match the else